pkg/apis/crd/v1alpha1/types.go

```diff
@@ -427,3 +355,122 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Namespace is the source Namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the source Pod.
+	Pod string `json:"pod"`
+	// IP is the source IPv4 or IPv6 address.
+	IP string `json:"ip"`
+}
+
+// Destination describes the destination spec of the PacketCapture.
+type Destination struct {
+	// Namespace is the destination Namespace.
+	Namespace string `json:"namespace"`
+	// Pod is the destination Pod, exclusive with destination Service.
+	Pod string `json:"pod"`
+	// Service is the destination Service, exclusive with destination Pod.
+	Service string `json:"service"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP string `json:"ip"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	UDP *UDPHeader `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP *TCPHeader `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	// IPFamily is the filter's ip family . default to `IPv4`.
+	IPFamily v1.IPFamily `json:"ipFamily"`
+	// Protocol represents the transport protocol. default to ICMP(1).
```

Comment:
We should list what protocol string values we support.

Reply:
you mean in the comments? Is that enough?

Reply:
updated.